lib/npg/samplesheet/auto.pm

```diff
@@ -141,9 +147,21 @@ sub process {
     my $id_run = $r->id_run;
     $self->log->info('Considering ' . join q[,],$id_run,$r->instrument->name);
 
-    my $ss = npg::samplesheet->new(
-      run => $r, mlwh_schema => $self->mlwh_schema
-    );
+    my $ss;
+    if ($self->instrument_format eq $MISEQ_INSTRUMENT_FORMAT) {
+      $ss = npg::samplesheet->new(
+        run => $r,
+        mlwh_schema => $self->mlwh_schema
+      );
+    } else {
+      $ss = npg::samplesheet::novaseq_xseries->new(
+        id_run => $id_run,
```

Comment:
Starting to feel wary about passing id_run and run in the constructor. It will be wasteful to let the lazy build query again to get the run. As I understand we also need id_run because that way things are compatible when passing from command line. Don't think it is wrong, it just feels strange. Maybe there is something else I'm missing.

Reply:
Fixed, see https://github.com/wtsi-npg/npg_tracking/pull/796/commits/9a2c9f0a29aeab37388b0716af7d057aaedea986